internal/dag/httpproxy_processor.go

```diff
@@ -1343,6 +1334,53 @@ func (p *HTTPProxyProcessor) computeSecureVirtualHostAuthorization(validCond *co
 	return true
 }
 
+func computeVirtualHostRateLimitPolicy(proxy *contour_api_v1.HTTPProxy, rateLimitService *contour_api_v1alpha1.RateLimitServiceConfig, validCond *contour_api_v1.DetailedCondition) *RateLimitPolicy {
```

Comment:
I wonder if breaking this into two functions one for local and one for global make it easier to reason about

Reply:
I tried to make it simpler. Let me know what do you think